operator/v1/types_vsphere_problem_detector.go

```diff
@@ -0,0 +1,71 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// VSphereProblemDetector is used to manage and report health of Openshift
+// cluster running in vSphere environment.
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VSphereProblemDetector objects allows configuration and health reporting of
+// Vsphere problem detector operator.
+type VSphereProblemDetector struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec VSphereProblemDetectorSpec `json:"spec"`
+
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status VSphereProblemDetectorStatus `json:"status"`
+}
+
+// VSphereProblemDetectorSpec is the desired configuration of vSphere problem detector operator.
+type VSphereProblemDetectorSpec struct {
+	OperatorSpec `json:",inline"`
+}
+
+// VSphereProblemDetectorStatus is the status of vSphere problem detector operator and
+// health of Openshift and vSphere integration.
+type VSphereProblemDetectorStatus struct {
+	OperatorStatus `json:",inline"`
+	// LastCheckTime reports timestamp of last check
+	LastCheckTime metav1.Time `json:"lastCheckTime,omitempty"`
+	// List of checks and their status
+	Checks []VSphereHealthCheck `json:"vSphereHealthCheck,omitempty"`
+}
+
+type VSphereHealthCheck struct {
+	Name    string             `json:"name"`
+	Result  VSphereCheckResult `json:"vSphereCheckResult"`
+	Message string             `json:"message"`
+}
+
+type VSphereCheckResult string
+
+var (
+	// VSphereCheckPassed indicates health check for vsphere and openshift has passed.
+	VSphereCheckPassed VSphereCheckResult = "Passed"
+	// VSphereCheckFailed indicates health check for vsphere and openshift has failed.
+	VSphereCheckFailed VSphereCheckResult = "Failed"
+	// VSphereCheckWarning indicates there were warnings while verifying vsphere and openshift
+	// integration.
+	VSphereCheckWarning VSphereCheckResult = "Warning"
```

Comment:
nit: since these are doc strings, it'd be nice to properly capitalize `vsphere and openshift`.

Reply:
hmm, in theory I agree but if it is alright - I am going to leave them alone. I don't think it should matter much in the end.

Reply:
None of these strings will show up in `oc explain`. Please add proper docs on the fields above instead.